torch/distributed/distributed_c10d.py

```diff
@@ -753,6 +784,61 @@ def recv(tensor,
         return src
 
 
+@contextlib.contextmanager
+def _group_manager(backend):
+    if backend == Backend.NCCL:
+        ProcessGroupNCCL.group_start()
+    try:
+        yield
+    finally:
+        if backend == Backend.NCCL:
+            ProcessGroupNCCL.group_end()
+
+
+def batch_isend_irecv(op_list,
+                      tensor_list,
+                      peer_list,
+                      group_list=None,
+                      tag_list=None):
```

Comment:
Feeling this may not be the most intrinsic way. Users should firstly have a series of p2p calls in mind. To use this batched interface, they have to split these p2p calls into op list, tensor list and peer list. Could we have a class object of for p2pOp, recording isend or irecv, peer, tensor, tag, and pass into this interface as a list of [p2pOp]

Reply:
Not sure if I get the new design?

Reply:
I like @jiayisuse's suggestion here and I think the design you outline above is a good approach.

Reply:
Let me add p2pOp.

Reply:
Let me add p2pOP.